hw/ip/otp_ctrl/doc/_index.md

```diff
@@ -621,50 +792,80 @@ See further below for a detailed [Memory Map]({{< relref "#direct-access-memory-
 If the region accessed has a 64bit access granule, the 64bit chunk of read data can be read from the {{< regref DIRECT_ACCESS_RDATA_0 >}} and {{< regref DIRECT_ACCESS_RDATA_1 >}} registers.
 7. Go back to 1. and repeat until all data has been read.
-8. Enable write protection of the DAI by setting {{< regref DIRECT_ACCESS_REGWEN >}} to 0x0.
+
+Note that the hardware will set {{< regref DIRECT_ACCESS_REGWEN >}} to 0x0 while an operation is pending in order to temporarily lock write access to the CSRs registers.
 
 ### Programming Sequence
 
 A typical programming sequence looks as follows:
 
-1. Unprotect the DAI by setting {{< regref DIRECT_ACCESS_REGWEN >}} to 0x1.
+1. Check whether the DAI is idle by reading the {{< regref STATUS >}} register.
 2. If the region to be accessed has a 32bit access granule, place a 32bit chunk of data into {{< regref DIRECT_ACCESS_WDATA_0 >}}.
 If the region to be accessed has a 64bit access granule, both the {{< regref DIRECT_ACCESS_WDATA_0 >}} and {{< regref DIRECT_ACCESS_WDATA_1 >}} registers have to be used.
 3. Write the byte address for the access to {{< regref DIRECT_ACCESS_ADDRESS >}}.
 Note that the address is aligned with the granule, meaning that either 2 or 3 LSBs of the address are ignored, depending on whether the access granule is 32 or 64bit.
 4. Trigger a write command by writing 0x2 to {{< regref DIRECT_ACCESS_CMD >}}.
-5. Poll the {{< regref STATUS >}} until the state goes back to IDLE or ERROR.
+5. Poll the {{< regref STATUS >}} until the DAI state goes back to idle.
 Alternatively, the `otp_operation_done` interrupt can be enabled up to notify the processor once an access has completed.
-6. If the operation has finished with ERROR status, additional handling is required (see [Section on Error handling]({{< relref "#error-handling" >}})).
+6. If the status register flags a DAI error, additional handling is required (see [Section on Error handling]({{< relref "#error-handling" >}})).
 7. Go back to 1. and repeat until all data has been written.
-8. Enable write protection of the DAI by setting {{< regref DIRECT_ACCESS_REGWEN >}} to 0x0.
+
+Note that the hardware will set {{< regref DIRECT_ACCESS_REGWEN >}} to 0x0 while an operation is pending in order to temporarily lock write access to the CSRs registers.
 
 ### Digest Calculation Sequence
 
 The hardware digest computation for the hardware and secret partitions can be triggered as follows:
 
-1. Unprotect the DAI by setting {{< regref DIRECT_ACCESS_REGWEN >}} to 0x1.
-3. Write the correct partition index to {{< regref DIRECT_ACCESS_ADDRESS >}}.
+1. Check whether the DAI is idle by reading the {{< regref STATUS >}} register.
+3. Write the partition base address to {{< regref DIRECT_ACCESS_ADDRESS >}}.
 4. Trigger a digest calculation command by writing 0x3 to {{< regref DIRECT_ACCESS_CMD >}}.
-5. Poll the {{< regref STATUS >}} until the state goes back to IDLE or ERROR.
+5. Poll the {{< regref STATUS >}} until the DAI state goes back to idle.
```

Comment:
I could not command line 831, but to write digest sequence, we are writing 0x4 instead of 0x3 i believe.

Reply:
good catch - fixed!